index.js

```diff
@@ -169,6 +281,212 @@ class Pawn extends Piece {
+
+    if (this.isValidSquare(...elSquare)) {
+      if (!this.isSquareOccupied(...elSquare)) {
+        arr.push(Move.fromSquare(elSquare, this));
+      } else if (
+        this.isSquareOccupied(...elSquare) &&
+        this.getSquareContent(...elSquare).color !== this.color
+      ) {
+        arr.push(
+          Move.fromSquare(elSquare, this, this.getSquareContent(...elSquare))
+        );
+      }
+    }
+    return arr;
+  }
+
+  get moves() {
+    const available = [];
+
+    this.elMove(this.row, this.file, available, 2, 1);
+    this.elMove(this.row, this.file, available, 1, 2);
+
+    this.elMove(this.row, this.file, available, -2, 1);
+    this.elMove(this.row, this.file, available, -1, 2);
+
+    this.elMove(this.row, this.file, available, -2, -1);
+    this.elMove(this.row, this.file, available, -1, -2);
+
+    this.elMove(this.row, this.file, available, 2, -1);
+    this.elMove(this.row, this.file, available, 1, -2);
+
+    return available;
+  }
+
+  onMove(move) {
+    console.log(move);
+  }
+
+  get icon() {
+    return this.isWhite() ? "♘" : "♞";
+  }
+}
+
+class Queen extends Piece {
+  constructor(game, board, color, row, file) {
+    super(game, board, color);
+    this.row = row;
+    this.file = file;
+  }
+
+  name = "queen";
+
+  get moves() {
+    const available = [];
+    this.verticalMove(this.row, this.file, available, "up");
+    this.verticalMove(this.row, this.file, available, "down");
+    this.horizontalMove(this.row, this.file, available, "right");
+    this.horizontalMove(this.row, this.file, available, "left");
+
+    this.moveDiagonally(this.row, this.file, available, "up", "right");
+    this.moveDiagonally(this.row, this.file, available, "down", "left");
+    this.moveDiagonally(this.row, this.file, available, "up", "left");
+    this.moveDiagonally(this.row, this.file, available, "down", "right");
+    return available;
+  }
+
+  onMove(move) {
+    console.log(move);
+  }
+
+  get icon() {
+    return this.isWhite ? "♕" : "♛";
+  }
+}
+
+class King extends Piece {
+  constructor(game, board, color, row, file) {
+    super(game, board, color);
+    this.row = row;
+    this.file = file;
+    this.hasMoved = false;
+    this.isChecked = false;
+  }
+
+  name = "King";
+
+  moveOneSquare(row, file, arr, vert, horiz) {
+    const nextSquare = [row + vert, file + horiz];
+    if (
+      this.isValidSquare(...nextSquare) &&
+      !this.isSquareOccupied(...nextSquare)
+    ) {
+      arr.push(Move.fromSquare(nextSquare, this));
+      this.isNotChecked();
+    } else if (
+      this.isValidSquare(...nextSquare) &&
+      this.isSquareOccupied(...nextSquare) &&
+      this.getSquareContent(...nextSquare).color !== this.color
+    ) {
+      arr.push(
+        Move.fromSquare(nextSquare, this, this.isSquareOccupied(...nextSquare))
+      );
+    }
+    return arr;
+  }
+
+  isCheck() {
+    this.isChecked = true;
+  }
+
+  isNotChecked() {
+    this.isChecked = false;
+  }
```

Comment:
oops, this is assignment. the `isChecked` function implies that it's a predicate function (a func that returns a boolean). If you wanted to mutate the checked property do this instead:

```suggestion
  setChecked() {
    this.checked = true;
  }
  
  setNotChecked() {
    this.checked = false;
  }
```